chain/ethereum/src/trigger.rs

```diff
@@ -291,8 +358,12 @@ impl EthereumTrigger {
                 Some(address)
             }
             EthereumTrigger::Call(call) => Some(&call.to),
-            EthereumTrigger::Log(log, _) => Some(&log.address),
-
+            EthereumTrigger::Log(log_ref) => match log_ref {
+                LogRef::FullLog(log, _) => Some(&log.address),
+                LogRef::LogPosition(index, receipt) => {
+                    Some(&receipt.logs.get(*index).unwrap().address)
+                }
+            },
```

Comment:
We should not be matching on `LogRef`, it needs an `address()` method. Same goes for `block_hash` and `block_number` above.